### ISSUE TITLE
restoring action sometimes fails from vscode, then on next run polling fails

### ISSUE THREAD

**Issue by alexkli** [apache/openwhisk-wskdebug#25]
Error on quitting debugger in vscode:

```
Completed activation 4bbde23ef51e4561bde23ef51e656189 in 0.529 sec
Shutting down...
Error while restoring original action: { OpenWhiskError: GET https://adobeioruntime.net/api/v1/namespaces/105979_45469/actions/NUI-313/auth-custom-worker_wskdebug_original Returned HTTP 404 (Not Found) --> "The requested resource does not exist."
    at Client.handleErrors (/usr/local/lib/node_modules/@openwhisk/wskdebug/node_modules/openwhisk/lib/client.js:224:11)
    at params.then.catch.err (/usr/local/lib/node_modules/@openwhisk/wskdebug/node_modules/openwhisk/lib/client.js:155:58)
    at process._tickCallback (internal/process/next_tick.js:68:7)
  name: 'OpenWhiskError',
  message:
   'GET https://adobeioruntime.net/api/v1/namespaces/105979_45469/actions/NUI-313/auth-custom-worker_wskdebug_original Returned HTTP 404 (Not Found) --> "The requested resource does not exist."',
  error:
   { code: 'pfx0I0yYNfT6NMU8wt8k4hjSX7iPpL3P',
     error: 'The requested resource does not exist.' },
  statusCode: 404 }
```

On next start it seems to not install the agent, and polling fails with an unexpected response (? - don't have logs captured).

**Comment by alexkli** [comment]
Another similar case, where wskdebug is running normally and then a new invocation is made and it starts to get those "System not ready, status is stopped":

```
... wskdebug running normally...

=> Run: echo COPY_YOUR_WSKPROP
COPY_YOUR_WSKPROP
=> Run: echo COPY_YOUR_WSKPROP
COPY_YOUR_WSKPROP
Activation: 2045f58cb1f9455e85f58cb1f9d55e16
2020-03-24T23:29:45.634899200Z Internal system error: System not ready, status is stopped.
Completed activation 2045f58cb1f9455e85f58cb1f9d55e16 in 0.007 sec
Activation: ca1bf18aafca48ac9bf18aafcad8acde
2020-03-24T23:29:46.555238200Z Internal system error: System not ready, status is stopped.
Completed activation ca1bf18aafca48ac9bf18aafcad8acde in 0.005 sec
Activation: aa337320ed884a71b37320ed889a7152
2020-03-24T23:29:47.682195300Z Internal system error: System not ready, status is stopped.
Completed activation aa337320ed884a71b37320ed889a7152 in 0.005 sec
Activation: 558faeb459ba45808faeb459bac58034
2020-03-24T23:29:48.899365900Z Internal system error: System not ready, status is stopped.
Completed activation 558faeb459ba45808faeb459bac58034 in 0.006 sec
Activation: d986a6e05df54c0786a6e05df59c074a
2020-03-24T23:29:50.537270500Z Internal system error: System not ready, status is stopped.
Completed activation d986a6e05df54c0786a6e05df59c074a in 0.005 sec
Activation: e5efd4b6288c44b5afd4b6288cf4b5dc
2020-03-24T23:29:52.934839500Z Internal system error: System not ready, status is stopped.
Completed activation e5efd4b6288c44b5afd4b6288cf4b5dc in 0.005 sec
Shutting down...
Shutting down...
Error while restoring original action: OpenWhiskError: GET https://adobeioruntime.net/api/v1/namespaces/105979_45469/actions/NUI-313/auth-custom-worker_wskdebug_original Returned HTTP 404 (Not Found) --> "The requested resource does not exist."
    at Client.handleErrors (/usr/local/lib/node_modules/@openwhisk/wskdebug/node_modules/openwhisk/lib/client.js:224:11)
    at /usr/local/lib/node_modules/@openwhisk/wskdebug/node_modules/openwhisk/lib/client.js:155:58
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async Debugger.restoreAction (/usr/local/lib/node_modules/@openwhisk/wskdebug/src/debugger.js:493:34)
    at async Debugger.tryCatch (/usr/local/lib/node_modules/@openwhisk/wskdebug/src/debugger.js:861:17)
    at async Debugger.shutdown (/usr/local/lib/node_modules/@openwhisk/wskdebug/src/debugger.js:196:13)
    at async Debugger.kill (/usr/local/lib/node_modules/@openwhisk/wskdebug/src/debugger.js:182:9)
    at async process.<anonymous> (/usr/local/lib/node_modules/@openwhisk/wskdebug/index.js:274:13) {
  name: 'OpenWhiskError',
  message: 'GET https://adobeioruntime.net/api/v1/namespaces/105979_45469/actions/NUI-313/auth-custom-worker_wskdebug_original Returned HTTP 404 (Not Found) --> "The requested resource does not exist."',
  error: {
    code: 'ukCFsWA82AplSkMp0JncAo5QAbQxeG6x',
    error: 'The requested resource does not exist.'
  },
  statusCode: 404
}
```

**Comment by alexkli** [comment]
Turns out there was a bug that on the next run _after_ a failed shutdown (e.g. VS code killed `wskdebug` before it completed restoring the original action and the agent was left behind in place of the actual action), it did things in the wrong order:

The agent was installed first, effectively doing a no op, since the agent was still there. Then it restored the original from the backup copy. And then it started to wait for invocations and expecting the agent to be there, while in fact the original action is present and `wskdebug` can't work properly.

Fixed in #40.
